pkg/networkpolicy/controller_test.go

```diff
@@ -92,27 +101,201 @@ func newController() *networkpolicyController {
 	npaClient := npaclientfake.NewSimpleClientset()
 	npaInformerFactory := npainformers.NewSharedInformerFactory(npaClient, 0)
 
-	controller := NewController(client,
-		&knftables.Fake{},
```

Comment:
Ah, that's the answer to "why is the knftables.Interface an argument".
(FTR, `&knftables.Fake{}` is wrong; you should use `knftables.NewFake()`)